Explain how to do local dev when you have DOs with a framework (#21198)
cloudflare/cloudflare-docs: src/content/docs/workers/wrangler/api.mdx

- [Workers AI bindings](/workers-ai/get-started/workers-wrangler/#2-connect-your-worker-to-workers-ai)

  <Render file="ai-local-usage-charges" product="workers" />

> - <a name="durable-objects"></a>[Durable Object bindings](/durable-objects/api/)

Reviewer: are we adding this link here just so we can specifically link to this section? feels a bit odd that DOs get this and other bindings not. Is linking to `Supported bindings` not enough?

Reviewer: Right, but again, the only reason we're adding it is because another page in our docs wants to link to that specific bit. So we are making decisions about how this page is semantically structured, based on how other docs pages interact with it, and that doesn't feel right. Even if we were to be consistent and convert all items in the bindings list into a URI fragments, that wouldn't make much sense in the context of this page either, since barely any of those items do anything else other than link to another page. DOs are the only exception, and that is because we are now making these changes. 

I won't die on this hill, and I understand that we are trying to link users to the relevant section with zero friction, but personally I don't think this is the way to go, for this particular case, for the reasons I've explained. I think linking to `Supported bindings` should be enough.

Author: got rid of the anchor link + updated the wrangler PR as well

  - To use a Durable Object binding with `getPlatformProxy`, always specify a [`script_name`](/workers/wrangler/configuration/#durable-objects).

>     For example, you might have the following binding in a file read by `getPlatformProxy`.

Reviewer: ```suggestion
    For example, you might have the following binding in a Wrangler configuration file read by `getPlatformProxy`.
```

    <WranglerConfig>

    ```toml
    [[durable_objects.bindings]]
    name = "MyDurableObject"
    class_name = "MyDurableObject"
    script_name = "external-do-worker"
    ```

    </WranglerConfig>

    You will need to declare your Durable Object `"MyDurableObject"` in another Worker, called `external-do-worker` in this example.

        ```ts title="./external-do-worker/src/index.ts"
        export class MyDurableObject extends DurableObject {
        	// Your DO code goes here
        }

        export default {
        	fetch() {
        			// Doesn't have to do anything, but a DO cannot be the default export
        			return new Response("Hello, world!");
        	},
        };
        ```

>         That Worker also needs a Wrangler config that looks like this:

Reviewer: ```suggestion
        That Worker also needs a Wrangler configuration file that looks like this:
```

        <WranglerConfig>
        ```json
        {
        	"name": "external-do-worker",
        	"main": "src/index.ts",
        	"compatibility_date": "XXXX-XX-XX"
        }
        ```
        </WranglerConfig>

>         If you are not using RPC with your Durable Object, you can run a separate Wrangler dev session alongside your framework development server.

Reviewer: ```suggestion
        If you are not using RPC with your Durable Object, you can run a separate `wrangler dev` session alongside your framework development server.
```
not 100% sure tho...

Author: Might be a pages dev command so I'll leave it as is

>         If you are not using RPC with your Durable Object, you can run a separate Wrangler dev session alongside your framework development server.

Reviewer: I feel like we might be confusing users with too many options here. Is there one option that works for both rpc/non-rpc?

Author: There's a trade off with each unfortunately. The first option lets you have hot reloading etc. by using the frameworks dev server, but no RPC. The second requires you build each time but allows RPC.

        Otherwise, you can build your application and run both Workers in the same Wrangler dev session.

>         If you are using Pages:

Reviewer: ```suggestion
        If you are using Pages run:
```

>         If you are using Pages:

Reviewer: do we really need to spell out how to run the dev cmd for Pages vs Workers + Assets? Feels a bit outside the scope of this docs. Maybe link to where this specific local dev doc lives (tried to find it but couldn't, but I know we have a section abt multi-worker local dev)?

Author: Yeah the problem is we don't have multi dev docs (@penalosa had a draft up but I'm not sure if there's a blocker). And the order you specify the configs in matters, so I thought it would be easier to just put it in.

        <PackageManagers type="exec" pkg="wrangler" args="pages dev -c path/to/pages/wrangler.jsonc -c path/to/external-do-worker/wrangler.jsonc" />

>         If you are using Workers with Assets:

Reviewer: ```suggestion
        If you are using Workers with Assets run:
```
